### PR TITLE
expose getContributionCount query, add hook

### PR DESCRIPTION
**Pull request by Flip-Liquid**
## Linear Ticket

[Pro-528](https://linear.app/govrn/issue/PRO-528/create-a-usedaocontributioncount-hook)

## Description

Add hook for useContributionCount

## Screencaptures

For frontend changes, considering adding a screenshot or screencapture of the changes.
Before/after screenshots may be helpful when relevant.

### CODE REVIEW

File: apps/protocol-frontend/src/hooks/useDaoContributionCount.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+const useDaoContributionCount = (args: {
+  startDate: Date;
+  endDate: Date;
+  guildIds?: number[] | null;
+  excludeUnassigned?: boolean;
+}) => {
+  const { userData, govrnProtocol: govrn } = useUser();
+
+  const { isLoading, isFetching, isError, error, data } = useQuery(
+    ['contributionGetCount', args],
```
Comment (alexkeating) [inline apps/protocol-frontend/src/hooks/useDaoContributionCount.ts:13]: This should probably be daoContributionGetCount now

File: apps/protocol-frontend/src/hooks/useDaoContributionCount.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import { useQuery } from '@tanstack/react-query';
+import { useUser } from '../contexts/UserContext';
+
+const useDaoContributionCount = (args: {
+  startDate: Date;
+  endDate: Date;
+  guildIds?: number[] | null;
```
Comment (alexkeating) [inline apps/protocol-frontend/src/hooks/useDaoContributionCount.ts:7]: This should be guildId not guildIds

File: apps/protocol-frontend/src/hooks/useDaoContributionCount.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  excludeUnassigned?: boolean;
+}) => {
+  const { userData, govrnProtocol: govrn } = useUser();
+
+  const { isLoading, isFetching, isError, error, data } = useQuery(
+    ['contributionGetCount', args],
+    async () => {
+      return await govrn.custom.getDaoContributionCount({
+        userId: userData?.id,
+        ...args,
```
Comment (alexkeating) [inline apps/protocol-frontend/src/hooks/useDaoContributionCount.ts:17]: Replace with guildId key

File: apps/protocol-frontend/src/hooks/useDaoContributionCount.ts (hunk 2 above)
Comment (jonathanprozzi) [reply apps/protocol-frontend/src/hooks/useDaoContributionCount.ts:7]: also won't be an array of numbers -- should likely just be a `number`

**Review by jonathanprozzi — COMMENTED** [review]
Small suggestions (also agree with Keating's) but this looks good. Thanks for the quick change after yesterday's sync!

File: apps/protocol-frontend/src/hooks/useDaoContributionCount.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  const { isLoading, isFetching, isError, error, data } = useQuery(
+    ['contributionGetCount', args],
+    async () => {
+      if (!userData?.id) {
+        console.error('getUserContributionsCount has no userData.id');
+        return [];
+      }
+
+      return await govrn.custom.getContributionCount({
+        userId: userData?.id,
```
[the remark this answers is not in the document]
Comment (Flip-Liquid) [reply apps/protocol-frontend/src/hooks/useDaoContributionCount.ts:16]:
```
{
  getDaoContributionCount(where: {
    guildId: 2, startDate: "2021-12-21 00:00:00.000", endDate: "2022-12-21 00:00:00.000"
  }) 
}
```

**Review by jonathanprozzi — COMMENTED** [review]
@Flip-Liquid I think that the types for `useDaoContributionCount` and `useContributionCount` may have been reversed -- `useContributionCount` is meant to take in the array of `guildIds` since it's used in the UserDashboard

**Comment by Flip-Liquid** [comment]
they absolutely were flipped 🤦 @jonathanprozzi . should be fixed now. tested the query again, and it looked like it was working from graphiql

**Pull request merged**